src/Toolkit/PubSub/V1/PublisherApi.php

```diff
@@ -0,0 +1,671 @@
+     *    @var int $timeoutMillis
+     *          Timeout to use for this call. Only used if $retrySettings
+     *          is not set.
+     * }
+     *
+     * @return google\pubsub\v1\PublishResponse
+     *
+     * @throws Google\GAX\ApiException if the remote call fails
+     */
+    public function publish($topic, $messages, $optionalArgs = [], $callSettings = [])
+    {
+        $request = new PublishRequest();
+        $request->setTopic($topic);
+        foreach ($messages as $elem) {
+            $request->addMessages($elem);
+        }
+
+        $mergedSettings = $this->defaultCallSettings['publish']->merge(
+            new CallSettings($callSettings)
+        );
+        $callable = ApiCallable::createApiCall(
+            $this->stub,
+            'Publish',
+            $mergedSettings,
+            $this->descriptors['publish']
+        );
+
+        return $callable(
+            $request,
+            [],
+            ['call_credentials_callback' => $this->createCredentialsCallback()]);
+    }
+
+    /**
+     * Gets the configuration of a topic.
+     *
+     * Sample code:
+     * ```
+     * try {
+     *     $publisherApi = new PublisherApi();
+     *     $formattedTopic = PublisherApi::formatTopicName("[PROJECT]", "[TOPIC]");
+     *     $response = $publisherApi->getTopic($formattedTopic);
+     * } finally {
+     *     if (isset($publisherApi)) {
+     *         $publisherApi->close();
+     *     }
+     * }
+     * ```
+     *
+     * @param string $topic        The name of the topic to get.
+     * @param array  $optionalArgs {
+     *                             Optional. There are no optional parameters for this method yet;
+     *                             this $optionalArgs parameter reserves a spot for future ones.
+     *                             }
+     * @param array  $callSettings {
+     *                             Optional.
+     *
+     *    @var Google\GAX\RetrySettings $retrySettings
+     *          Retry settings to use for this call. If present, then
+     *          $timeout is ignored.
+     *    @var int $timeoutMillis
+     *          Timeout to use for this call. Only used if $retrySettings
+     *          is not set.
+     * }
+     *
+     * @return google\pubsub\v1\Topic
+     *
+     * @throws Google\GAX\ApiException if the remote call fails
+     */
+    public function getTopic($topic, $optionalArgs = [], $callSettings = [])
+    {
+        $request = new GetTopicRequest();
+        $request->setTopic($topic);
+
+        $mergedSettings = $this->defaultCallSettings['getTopic']->merge(
+            new CallSettings($callSettings)
+        );
+        $callable = ApiCallable::createApiCall(
+            $this->stub,
+            'GetTopic',
+            $mergedSettings,
+            $this->descriptors['getTopic']
+        );
+
+        return $callable(
+            $request,
+            [],
+            ['call_credentials_callback' => $this->createCredentialsCallback()]);
```

Comment:
Where does this get called?  And why are we using `updateMetadata` rather than `fetchAuthToken`  in [GrpcBootstrap](https://github.com/googleapis/gax-php/blob/640b9341cc4852ce7b0eea7b2bc85ebb90fe5c43/src/GrpcBootstrap.php#L54)?